back/engines/commercial/report_builder/spec/services/report_builder/queries/analytics/active_users_spec.rb

```diff
@@ -6,25 +6,81 @@
   subject(:query) { described_class.new(build(:user)) }
 
   describe '#run_query' do
-    let(:date) { Date.new(2022, 9, 1) }
-    let(:idea) do
+    it 'returns active users' do
+      date = Date.new(2022, 9, 1)
+      project = create(:single_phase_ideation_project)
       create(:dimension_date, date: date)
       create(:dimension_type, name: 'idea', parent: 'post')
+      create(:idea, created_at: date, project: project)
 
-      create(:idea, created_at: date)
-    end
-
-    it 'returns active users' do
-      params = { start_at: date - 1.day, end_at: date + 1.day, project_id: idea.project_id }
+      params = { start_at: date - 1.day, end_at: date + 1.day, project_id: project.id }
       expect(query.run_query(**params)).to eq(
         [
           [{
-            'count_dimension_user_id' => 1,
+            'count_participant_id' => 1,
             'dimension_date_created.month' => '2022-09',
             'first_dimension_date_created_date' => date
           }],
           [{
-            'count_dimension_user_id' => 1
+            'count_participant_id' => 1
+          }]
+        ]
+      )
+    end
+
+    it 'correctly deals with anonymous and deleted users' do
+      project = create(:project)
+      pp1, pp2, pp3, pp4 = create_list(:user, 4)
+      date_september = Date.new(2022, 9, 10)
+      date_october = Date.new(2022, 10, 5)
+
+      create(:dimension_date, date: date_september)
+      create(:dimension_date, date: date_october)
+      create(:dimension_type, name: 'idea', parent: 'post')
+      create(:dimension_type, name: 'comment', parent: 'idea')
+
+      ### SEPTEMBER ###
+      # Create a bunch of ideas and comments with users (4 participants)
+      idea1 = create(:idea, project: project, author: pp1, created_at: date_september) # 1
+      idea2 = create(:idea, project: project, author: pp2, created_at: date_september) # 2
+      create(:comment, post: idea1, author: pp3, created_at: date_september) # 2
+      create(:idea, project: project, created_at: date_september) # 4
+      create(:comment, post: idea2, author: pp1, created_at: date_september)
```

Comment:
The meaning of those comments is not very clear to me, or should the third one by a 3. In any case, I would suggest elaborating them a bit or removing them altogether.